superset/assets/src/explore/exploreUtils.js

```diff
@@ -1,11 +1,22 @@
 /* eslint camelcase: 0 */
 import URI from 'urijs';
+import { availableDomains } from '../utils/hostNamesConfig';
 
 export function getChartKey(explore) {
   const slice = explore.slice;
   return slice ? (slice.slice_id) : 0;
 }
 
+let requestCounter = 0;
+function getHostName(allowDomainSharding = false) {
+  let currentIndex = 0;
+  if (allowDomainSharding) {
+    currentIndex = requestCounter++ % availableDomains.length;
```

Comment:
nit -- in many repos `++` violates a [lint rule](https://eslint.org/docs/rules/no-plusplus), I also think `requestCounter  += 1;` is more readable

Reply:
fixed.